### PR TITLE
Fix: build failed on LoongArch

### PR DESCRIPTION
**Pull request by yetist**
According to the documentation at https://github.com/loongson/la-abi-specs/blob/release/lapcs.adoc#the-base-abi-variants, valid -mabi parameters are lp64s, lp64f, lp64d, ilp32s, ilp32f and ilp32d.

### CODE REVIEW

File: Makefile.system [hunk 1]
```diff
@@ -954,7 +954,7 @@ endif
 ifeq ($(ARCH), loongarch64)
 LA64_ABI=$(shell $(CC) -mabi=lp64d -c $(TOPDIR)/cpuid_loongarch64.c -o /dev/null > /dev/null 2> /dev/null && echo lp64d)
 ifneq ($(LA64_ABI), lp64d)
-LA64_ABI=lp64
+LA64_ABI=lp64d
```
Comment (martin-frbg) [inline Makefile.system]: sorry, this change is almost certainly incorrect - the code here actually checks if the compiler supports the lp64d ABI and falls back to the older lp64 name if the compiler is too old to accept -mabi=lp64d

**Comment by martin-frbg** [comment]
Thanks - as far as I know, there may still be older versions of the Loongson gcc in use that do not understand the new ABI name. @XiWeiGu addressed this in #4077 for the Makefile build - the corresponding cmake files indeed appear to be outdated, but I wonder if a similar compatibility check would be needed there as well.

File: Makefile.system (hunk 1 above)
Comment (yetist) [reply Makefile.system]: ok, done.

**Comment by martin-frbg** [comment]
3R5 and 2K1000 targets appear to be broken by the new compiler in CI ?

**Comment by yetist** [comment]
> 3R5 and 2K1000 targets appear to be broken by the new compiler in CI ?

Maybe the new toolchain enable SIMD, and the kernel or qemu does not support SIMD?

@sunhaiyong1978 @yinshiyou

File: .github/workflows/loongarch64.yml [hunk 2]
```diff
@@ -40,8 +40,8 @@ jobs:
 
       - name: Download and install loongarch64-toolchain
         run: |
-          wget https://github.com/loongson/build-tools/releases/download/2022.09.06/loongarch64-clfs-7.3-cross-tools-gcc-glibc.tar.xz
-          tar -xf loongarch64-clfs-7.3-cross-tools-gcc-glibc.tar.xz -C /opt
+          wget https://github.com/loongson/build-tools/releases/download/2023.08.08/CLFS-loongarch64-8.1-x86_64-cross-tools-gcc-glibc.tar.xz
+          tar -xf CLFS-loongarch64-8.1-x86_64-cross-tools-gcc-glibc.tar.xz -C /opt
```
Comment (yinshiyou) [inline .github/workflows/loongarch64.yml:44]: Add "DYNAMIC_ARCH=1" into opts at line 19,22,25,28

**Comment by yinshiyou** [comment]
> > 3R5 and 2K1000 targets appear to be broken by the new compiler in CI ?
> 
> Maybe the new toolchain enable SIMD, and the kernel or qemu does not support SIMD?
> 
> @sunhaiyong1978 @yinshiyou

You updated the cross compiler which support SIMD, now qemu report "Illegal instruction", it means qemu don't support SIMD and the runtime check failed. The runtime check should be enabled in CI, you can add "DYNAMIC_ARCH=1" in .github/workflows/loongarch64.yml.

File: cmake/cc.cmake [hunk 3]
```diff
@@ -36,9 +36,9 @@ if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_C_COMPILER_ID} STREQUAL "LS
 
     if (LOONGARCH64)
       if (BINARY64)
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp64d")
       else ()
-        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=lp32")
+        set(CCOMMON_OPT "${CCOMMON_OPT} -mabi=ilp32d")
```
Comment (yinshiyou) [inline cmake/cc.cmake]: It's better to add a compiler check here? use new abi from a certain version.

File: .github/workflows/loongarch64.yml (hunk 2 above)
Comment (yetist) [reply .github/workflows/loongarch64.yml:44]: Thank you, it has been added.

File: cmake/cc.cmake (hunk 3 above)
Comment (yetist) [reply cmake/cc.cmake]: Done, waiting for the CI results.

File: cmake/fc.cmake
Comment (martin-frbg) [inline cmake/fc.cmake:1]: I think we'd want to use the compiler check here too ?

Comment (yetist) [reply cmake/fc.cmake:1]: Done.

Comment (martin-frbg) [reply cmake/fc.cmake:1]: thankyou

**Pull request merged**